### PR TITLE
Added Job listing page

### LINKED ISSUE #60: [FEATURE] Develop and Implement Search and Filter Functionalities on Careers Page
Develop the necessary components and implement functionalities for searching and filtering job listings on the careers page. The search functionality should include suggested keywords as the user types and allow searching by location with suggested keywords displayed in a dropdown. Integrate these functionalities with the appropriate API endpoints to ensure real-time and accurate search results.

**Purpose**
Users should be able to search and filter job listings on the careers page by categories and locations. This functionality will help users find relevant job listings more efficiently by using suggested keywords and integrated search capabilities.

### Acceptable Criteria
**Requirements**

- Search Bar with Suggested Keywords:
  - Develop a search bar that suggests keywords for job titles and categories as the user types.
  - Integrate the search bar with the appropriate API endpoint to fetch and display suggested keywords.
- Location Search Bar with Suggested Keywords:
  - Develop a location search bar that suggests keywords for locations as the user types.
  - Integrate the location search bar with the appropriate API endpoint to fetch and display suggested locations.
- Implement search functionality to find job listings by title, category, and location  with the appropriate API endpoints.
- Implement filter functionality to refine job listings by categories  with the appropriate API endpoints.
- Ensure the search and filter functionalities update the displayed job listings in real-time.
- Ensure the search bar, location search bar, and filter components are responsive and adapt well to different screen sizes
- Ensure the components and functionalities match the provided design specifications.

**Tasks**

- [ ] Develop a search bar for job titles and categories with suggested keywords displayed in a dropdown.
- [ ]  Develop a location search bar with suggested keywords displayed in a dropdown
- [ ] Implement search functionality to find job listings by title and location.
- [ ]  Implement filter functionality to refine job listings by categories.
- [ ] Integrate the search bar and location search bar with the appropriate API endpoint to fetch suggested keywords.
- [ ] Ensure the search bar, location search bar, and filter components are responsive
- [ ]  Ensure consistency across browsers and devices.

Deadline: Friday 10pm

**Note**

- Ensure code quality through proper code structure and comments.
- Provide clear documentation on component usage and integration.
- Ensure smooth integration with existing components on the careers page.

Images and Figma Links
https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=417-23283&m=dev
image

### PR DESCRIPTION
**Pull request by Temiloluwagboyega**
# Description

closes #60

# Changes proposed

## What were you told to do?

I was told to create a job listing page, Develop the necessary components and implement functionalities for searching and filtering job listings on the careers page. The search functionality should include suggested keywords as the user types and allow searching by location with suggested keywords displayed in a dropdown. Integrate these functionalities with the appropriate API endpoints to ensure real-time and accurate search results.

## What did you do?

i created 3 components, for the nav, the job listing component anffd the footer

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x ] My code follows the code style of this project.
- [x ] This PR does not contain plagiarized content.
- [x ] The title and description of the PR is clear and explains the approach.
- [x ] I am making a pull request against the **dev branch** (left side).
- [x ] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [ x] I am only making changes to files I was requested to.

# Screenshots/Videos

Screenshot_21-7-2024_1320_localhost

https://github.com/user-attachments/assets/181d68cf-1060-4107-873c-3b344cc06399

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +21,7 @@
     "clsx": "^2.1.1",
     "isbot": "^4.1.0",
     "lucide-react": "^0.408.0",
+    "pnpm": "^9.5.0",
```
Comment (abiona01) [inline package.json:24]: I beg all available pardons????

File: package.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
     "typecheck": "tsc"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-free": "^6.6.0",
```
Comment (abiona01) [inline package.json:14]: Ehh why?

File: vite.config.ts.timestamp-1721428823603-752cbcb748d68.mjs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// vite.config.ts
+import { vitePlugin as remix } from "file:///C:/Users/temil/Desktop/Temi's%20projects/HNG/hng_boilerplate_remix/node_modules/@remix-run/dev/dist/index.js";
+import { defineConfig } from "file:///C:/Users/temil/Desktop/Temi's%20projects/HNG/hng_boilerplate_remix/node_modules/vite/dist/node/index.js";
+import tsconfigPaths from "file:///C:/Users/temil/Desktop/Temi's%20projects/HNG/hng_boilerplate_remix/node_modules/vite-tsconfig-paths/dist/index.mjs";
+var vite_config_default = defineConfig({
```
Comment (abiona01) [inline vite.config.ts.timestamp-1721428823603-752cbcb748d68.mjs:5]: I am very confused. I don't understand what's going on. Please explain

File: vite.config.ts.timestamp-1721428823603-752cbcb748d68.mjs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
...cm9qZWN0c1xcXFxITkdcXFxcaG5nX2JvaWxlcnBsYXRlX3JlbWl4XFxcXHZpdGUuY29uZmlnLnRzXCI7Y29uc3QgX192aXRlX2luamVjdGVkX29yaWdpbmFsX2ltcG9ydF9tZXRhX3VybCA9IFwiZmlsZTovLy9DOi9Vc2Vycy90ZW1pbC9EZXNrdG9wL1RlbWkncyUyMHByb2plY3RzL0hORy9obmdfYm9pbGVycGxhdGVfcmVtaXgvdml0ZS5jb25maWcudHNcIjtpbXBvcnQgeyB2aXRlUGx1Z2luIGFzIHJlbWl4IH0gZnJvbSBcIkByZW1peC1ydW4vZGV2XCI7XHJcbmltcG9ydCB7IGRlZmluZUNvbmZpZyB9IGZyb20gXCJ2aXRlXCI7XHJcbmltcG9ydCB0c2NvbmZpZ1BhdGhzIGZyb20gXCJ2aXRlLXRzY29uZmlnLXBhdGhzXCI7XHJcblxyXG5leHBvcnQgZGVmYXVsdCBkZWZpbmVDb25maWcoe1xyXG4gIHBsdWdpbnM6IFtcclxuICAgIHJlbWl4KHtcclxuICAgICAgZnV0dXJlOiB7XHJcbiAgICAgICAgdjNfZmV0Y2hlclBlcnNpc3Q6IHRydWUsXHJcbiAgICAgICAgdjNfcmVsYXRpdmVTcGxhdFBhdGg6IHRydWUsXHJcbiAgICAgICAgdjNfdGhyb3dBYm9ydFJlYXNvbjogdHJ1ZSxcclxuICAgICAgfSxcclxuICAgIH0pLFxyXG4gICAgdHNjb25maWdQYXRocygpLFxyXG4gIF0sXHJcbn0pO1xyXG4iXSwKICAibWFwcGluZ3MiOiAiO0FBQWtZLFNBQVMsY0FBYyxhQUFhO0FBQ3RhLFNBQVMsb0JBQW9CO0FBQzdCLE9BQU8sbUJBQW1CO0FBRTFCLElBQU8sc0JBQVEsYUFBYTtBQUFBLEVBQzFCLFNBQVM7QUFBQSxJQUNQLE1BQU07QUFBQSxNQUNKLFFBQVE7QUFBQSxRQUNOLG1CQUFtQjtBQUFBLFFBQ25CLHNCQUFzQjtBQUFBLFFBQ3RCLHFCQUFxQjtBQUFBLE1BQ3ZCO0FBQUEsSUFDRixDQUFDO0FBQUEsSUFDRCxjQUFjO0FBQUEsRUFDaEI7QUFDRixDQUFDOyIsCiAgIm5hbWVzIjogW10KfQo=
```
Comment (abiona01) [inline vite.config.ts.timestamp-1721428823603-752cbcb748d68.mjs:20]: Remove comment

**Review by abiona01 — CHANGES_REQUESTED** [review]

File: vite.config.ts.timestamp-1721428823603-752cbcb748d68.mjs [hunk 5]
```diff
@@ -0,0 +1,20 @@
+// vite.config.ts
+import { vitePlugin as remix } from "file:///C:/Users/temil/Desktop/Temi's%20projects/HNG/hng_boilerplate_remix/node_modules/@remix-run/dev/dist/index.js";
+import { defineConfig } from "file:///C:/Users/temil/Desktop/Temi's%20projects/HNG/hng_boilerplate_remix/node_modules/vite/dist/node/index.js";
+import tsconfigPaths from "file:///C:/Users/temil/Desktop/Temi's%20projects/HNG/hng_boilerplate_remix/node_modules/vite-tsconfig-paths/dist/index.mjs";
```
Comment (abiona01) [inline vite.config.ts.timestamp-1721428823603-752cbcb748d68.mjs:4]: What is going on????

File: vite.config.ts.timestamp-1721428823603-752cbcb748d68.mjs (hunk 4 above)
Comment (abiona01) [inline vite.config.ts.timestamp-1721428823603-752cbcb748d68.mjs:20]: remove comment

File: app/components/ui/jobListing.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+  { title: "Sales Executive", location: "Remote", type: "Full-time", description: "The Sales Executive will drive sales growth by identifying and pursuing new business opportunities...", salary: "$4000 - $6500/month" },
+  { title: "Operations Manager", location: "Lagos, Nigeria", type: "Full-time", description: "As an Operations Manager, you will oversee the day-to-day operations of the company...", salary: "$6500 - $9000/month" },
+  { title: "Customer Support Specialist", location: "Remote", type: "Full-time", description: "The Customer Support Specialist will assist customers with their inquiries and resolve any issues they may have...", salary: "$3500 - $5500/month" },
+  { title: "Network Administrator", location: "Lagos, Nigeria", type: "Full-time", description: "The Network Administrator will manage and maintain the company's network infrastructure...", salary: "$5000 - $7000/month" },
+  { title: "Project Manager", location: "Remote", type: "Full-time", description: "The Project Manager will oversee and manage projects from inception to completion...", salary: "$6000 - $8500/month" },
```
Comment (abiona01) [inline app/components/ui/jobListing.tsx:16]: Isn't there a folder for data??

**Review by abiona01 — CHANGES_REQUESTED** [review]

File: vite.config.ts.timestamp-1721428823603-752cbcb748d68.mjs (hunk 5 above)
Comment (Temiloluwagboyega) [reply vite.config.ts.timestamp-1721428823603-752cbcb748d68.mjs:4]: will resolve it, am so sorry. my laptop has been switched off

**Pull request closed without merging**